src/Http/Controllers/CP/Collections/EntriesController.php

```diff
@@ -28,6 +28,10 @@ public function index(FilteredRequest $request, $collection)
     {
         $this->authorize('view', $collection);
 
+        if ($response = $this->ensureCollectionIsAvailableOnSite($collection, Site::current())) {
```

Comment:
1. This controller method is for the ajax call so a redirect wouldn't make sense. This probably should go into `CollectionController@show`.
2. You want `Site::selected()`

Reply:
Thanks, I've moved and updated that